pkg/targets/timescaledb/process.go

```diff
@@ -219,17 +222,34 @@ func (p *processor) processCSI(hypertable string, rows []*insertData) uint64 {
 	}
 	cols = append(cols, tableCols[hypertable]...)
 
-	if p.opts.ForceTextFormat {
-		tx := MustBegin(p._db)
-		stmt, err := tx.Prepare(pq.CopyIn(hypertable, cols...))
+	var stmt *sql.Stmt
+	var err error
+
+	if p.opts.UseCopy && !p.opts.ForceTextFormat {
+		rows := pgx.CopyFromRows(dataRows)
+		inserted, err := p._pgxConn.CopyFrom(context.Background(), pgx.Identifier{hypertable}, cols, rows)
 		if err != nil {
 			panic(err)
 		}
+		if inserted != int64(len(dataRows)) {
+			fmt.Fprintf(os.Stderr, "Failed to insert all the data! Expected: %d, Got: %d", len(dataRows), inserted)
+			os.Exit(1)
+		}
+	} else {
+		tx := MustBegin(p._db)
+		if p.opts.UseCopy {
+			stmt, err = tx.Prepare(pq.CopyIn(hypertable, cols...))
+			for _, r := range dataRows {
+				stmt.Exec(r...)
+			}
+			_, err = stmt.Exec()
+		} else {
+			stmtString := genBatchInsertStmt(hypertable, cols, len(dataRows))
```

Comment:
do we still need this now that multinode supports copy?

Reply:
Good question. I actually wasn't sure about this either, especially after it bit me the first time I tried to run it and didn't have the config setup correctly.

@erimatnor or @niksajakovljevic - any thoughts?

Reply:
If I am not mistaken, I think TSBS at some point switched from COPY to INSERTs (this wasn't multi-node related). I think we just wanted to retain the option to benchmark also the INSERT path. Some people cannot use COPY because it doesn't support upserts, etc.